singer_sdk/streams/sql.py

```diff
@@ -825,9 +830,33 @@ def _adapt_column_type(
         Raises:
             NotImplementedError: if altering columns is not supported.
         """
-        current_type = self._get_column_type(full_table_name, column_name)
+        current_type: sqlalchemy.types.TypeEngine = self._get_column_type(
+            full_table_name, column_name
+        )
+
+        # Check if the existing column type and the sql type are the same
+        if isinstance(sql_type, type(current_type)):
```

Comment:
Played with this a little bit

```
 if isinstance(sql_type, type(current_type)):
 ```
 
 ```
 (Pdb) print(f"{sql_type=},{str(sql_type)=}. {current_type=}, {str(current_type)=}")
sql_type=VARCHAR(),str(sql_type)='VARCHAR'. current_type=VARCHAR(), str(current_type)='VARCHAR'
(Pdb) print(type(sql_type))
<class 'sqlalchemy.sql.sqltypes.VARCHAR'>
```
This will work for our "generic" types of VARCHAR. If we get to caring about varchar(6) vs varchar(1000) and needing to change this will present an issue but for the generic SDK this might be alright! 

This is where `str(sql_type)` might be better. Ideally `__eq__` in this case would work for us, but SQLAlchemy overrides that for some other purposes (I'm not well versed enough in SQLAlchemy to understand it all yet)

Reply:
I did have `type(sql_type) == type(current_type)` initally but flake8 during pre-commit checks asked me to use `isinstance` when comparing class types.  I didn't think to convert it to a str().  I am not sure about varchar(6) vs varchar(1000).  I will have to work with those a little to see.